app/source/js/viewmodels/AppointmentCard.js

```diff
@@ -311,34 +315,17 @@ function AppointmentCardViewModel(params) {
         app.modals.showNotification({
             title: 'Done!',
             message: msg
-        });
+        })
+        .then(function() {
+            this.emit(saveEvent, this.sourceItem());
+        }.bind(this));
     }.bind(this);
 
     this.cancelBookingByServiceProfessional = function() {
         if (!this.bookingCanBeCancelledByServiceProfessional()) return;
         this.isSaving(true);
         app.model.bookings.cancelBookingByServiceProfessional(this.bookingID())
-        .then(afterSaveBooking)
-        .catch(function(err) {
```

Comment:
It seems like, because of removal of cancelBookingByClient (L337), this error management lines gets replaced by the ones from that method. If cancelBookingByClient is restored, this lines must too (to don't forget the error management).